addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -339,254 +280,151 @@ private String getFormattedParameters(Collection<String> stateNames) {
-        URL cookieUrl = new URL(url);
-        HttpsURLConnection connection = (HttpsURLConnection) cookieUrl.openConnection();
-        connection.setDoOutput(false);
-        connection.setRequestMethod("GET");
-        setConnectionDefaults(connection);
-        connection.setRequestProperty("Cookie", cookie);
+    private String sendDataToTahomaWithCookie(String url, String urlParameters) throws InterruptedException, ExecutionException, TimeoutException, SomfyTahomaException {
+        logger.trace("Sending POST to Tahoma to url: {} with data: {}", url, urlParameters);
 
-        return connection.getInputStream();
-    }
+        ContentResponse response = httpClient.newRequest(url)
+                .method(HttpMethod.POST)
+                .header(HttpHeader.ACCEPT_LANGUAGE, "en-US,en")
+                .header(HttpHeader.ACCEPT_ENCODING, "gzip, deflate")
+                .header("X-Requested-With", "XMLHttpRequest")
+                .agent(TAHOMA_AGENT)
+                .content(new StringContentProvider(urlParameters))
+                .send();
 
-    public InputStream sendDataToTahomaWithCookie(String url, String postData) throws Exception {
-        logger.trace("Sending POST to Tahoma to url: {} with data: {}", url, postData);
-        URL cookieUrl = new URL(url);
-        HttpsURLConnection connection = (HttpsURLConnection) cookieUrl.openConnection();
-        connection.setDoOutput(true);
-        connection.setRequestMethod("POST");
-        setConnectionDefaults(connection);
-        connection.setRequestProperty("Content-Length", Integer.toString(postData.length()));
-        connection.setRequestProperty("Cookie", cookie);
-        try (DataOutputStream wr = new DataOutputStream(connection.getOutputStream())) {
-            wr.write(postData.getBytes(StandardCharsets.UTF_8));
+        if (response.getStatus() < 200 || response.getStatus() >= 300) {
+            logger.error("Received error code: {}", response.getStatus());
+            if (response.getStatus() == 404) {
+                throw new SomfyTahomaException("Not logged in");
+            }
         }
-
-        return connection.getInputStream();
+        return response.getContentAsString();
     }
 
-    public void setConnectionDefaults(HttpsURLConnection connection) {
-        connection.setInstanceFollowRedirects(false);
-        connection.setRequestProperty("User-Agent", TAHOMA_AGENT);
-        connection.setRequestProperty("Accept-Language", "en-US");
-        connection.setRequestProperty("Accept-Encoding", "gzip, deflate");
-        connection.setUseCaches(false);
-    }
+    private String sendDeleteToTahomaWithCookie(String url) throws InterruptedException, ExecutionException, TimeoutException, SomfyTahomaException {
+        logger.trace("Sending DELETE to Tahoma to url: {}", url);
 
-    public InputStream sendDeleteToTahomaWithCookie(String url) throws Exception {
+        ContentResponse response = httpClient.newRequest(url)
```

Comment:
You create this request quite often, how about factoring it out into a method that takes the url as an argument? On the result you can call `send()` here.